src/main/java/org/zalando/nakadi/controller/CursorOperationsController.java

```diff
@@ -0,0 +1,202 @@
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.context.request.NativeWebRequest;
+import org.zalando.nakadi.domain.NakadiCursor;
+import org.zalando.nakadi.domain.NakadiCursorDistanceQuery;
+import org.zalando.nakadi.domain.NakadiCursorDistanceResult;
+import org.zalando.nakadi.domain.NakadiCursorLag;
+import org.zalando.nakadi.domain.ShiftedNakadiCursor;
+import org.zalando.nakadi.exceptions.InvalidCursorException;
+import org.zalando.nakadi.exceptions.NakadiException;
+import org.zalando.nakadi.exceptions.runtime.InvalidCursorOperation;
+import org.zalando.nakadi.exceptions.runtime.MyNakadiRuntimeException1;
+import org.zalando.nakadi.problem.ValidationProblem;
+import org.zalando.nakadi.service.CursorConverter;
+import org.zalando.nakadi.service.CursorOperationsService;
+import org.zalando.nakadi.view.Cursor;
+import org.zalando.nakadi.view.CursorDistanceQuery;
+import org.zalando.nakadi.view.CursorDistanceResult;
+import org.zalando.nakadi.view.CursorLag;
+import org.zalando.nakadi.view.ShiftedCursor;
+import org.zalando.problem.MoreStatus;
+import org.zalando.problem.Problem;
+import org.zalando.problem.spring.web.advice.Responses;
+
+import javax.validation.Valid;
+import java.util.List;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import static org.springframework.http.HttpStatus.OK;
+import static org.springframework.http.ResponseEntity.status;
+
+@RestController
+public class CursorOperationsController {
+
+    private static final Logger LOG = LoggerFactory.getLogger(CursorOperationsController.class);
+
+    private final CursorConverter cursorConverter;
+    private final CursorOperationsService cursorOperationsService;
+
+    @Autowired
+    public CursorOperationsController(final CursorOperationsService cursorOperationsService,
+                             final CursorConverter cursorConverter) {
+        this.cursorOperationsService = cursorOperationsService;
+        this.cursorConverter = cursorConverter;
+    }
+
+    @RequestMapping(path = "/event-types/{eventTypeName}/cursor-distances", method = RequestMethod.POST)
+    public ResponseEntity<?> getDistance(@PathVariable("eventTypeName") final String eventTypeName,
+                                         @Valid @RequestBody final List<CursorDistanceQuery> queries,
+                                         final Errors errors,
+                                         final NativeWebRequest request) {
+
+        if (errors.hasErrors()) {
+            return Responses.create(new ValidationProblem(errors), request);
+        }
+
+        final List<NakadiCursorDistanceQuery> domainQueries = queries.stream()
+                .map(this.toCursorDistanceQuery(eventTypeName))
+                .collect(Collectors.toList());
+
+        final List<NakadiCursorDistanceResult> domainResult = cursorOperationsService.calculateDistance(domainQueries);
```

Comment:
For me it seems that there is a lot of unneeded complex things here.
What is the purpose of class NakadiCursorDistanceQuery?
What is the purpose of class NakadiCursorDistanceResult?

I thought that we can work within our domain model:
```
long distance = cursorOperationsService.calculateDistance(NakadiCursor, NakadiCursor)
```